examples/cpp/dds/RequestReplyExample/CalculatorServer.cpp

```diff
@@ -0,0 +1,224 @@
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastrtps/subscriber/SampleInfo.h>
+#include "CalculatorPubSubTypes.h"
+
+#include <vector>
+#include <mutex>
+#include <condition_variable>
+
+class CalculatorServer
+{
+    class Listener : public eprosima::fastdds::dds::DataReaderListener
+    {
+    public:
+
+        Listener(
+                eprosima::fastdds::dds::DataWriter* writer)
+            : writer_(writer)
+        {
+        }
+
+        void on_data_available(
+                eprosima::fastdds::dds::DataReader* reader) override
+        {
+            RequestType request;
+            eprosima::fastdds::dds::SampleInfo sample_info;
+
+            reader->take_next_sample(&request, &sample_info);
+
+            if (eprosima::fastdds::dds::InstanceStateKind::ALIVE_INSTANCE_STATE == sample_info.instance_state)
+            {
+                ReplyType reply;
+
+                if (OperationType::ADDITION == request.operation())
+                {
+                    reply.z(request.x() + request.y());
+                }
+                else if (OperationType::SUBSTRACTION == request.operation())
+                {
+                    reply.z(request.x() - request.y());
+                }
+                else if (OperationType::MULTIPLICATION == request.operation())
+                {
+                    reply.z(request.x() * request.y());
+                }
+                else if (OperationType::DIVISION == request.operation())
+                {
+                    if (0 != request.y())
+                    {
+                        reply.z(request.x() / request.y());
+                    }
+                }
+                eprosima::fastrtps::rtps::WriteParams write_params;
+                write_params.related_sample_identity().writer_guid(sample_info.sample_identity.writer_guid());
+                write_params.related_sample_identity().sequence_number(sample_info.sample_identity.sequence_number());
+                writer_->write(reinterpret_cast<void*>(&reply), write_params);
+            }
+        }
+
+    private:
+
+        eprosima::fastdds::dds::DataWriter* writer_ = nullptr;
+
+    };
+
+public:
+
+    bool init()
+    {
+        eprosima::fastdds::dds::DomainParticipantQos participant_qos;
+        eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->get_default_participant_qos(participant_qos);
+
```

Comment:
This could be omitted and `PARTICIPANT_QOS_DEFAULT` constant used instead.

Reply:
I've used this to allow us (or a user) to use an profile xml file with a participant_qos with `is_default_qos`.